server/src/main/java/io/crate/expression/scalar/arithmetic/SignFunction.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar.arithmetic;
+
+import static io.crate.metadata.functions.Signature.scalar;
+
+import io.crate.expression.scalar.UnaryScalar;
+import io.crate.metadata.Functions;
+import io.crate.metadata.Scalar;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+public class SignFunction {
+
+    public static final String NAME = "sign";
+
+    public static void register(Functions.Builder builder) {
+        for (var type : DataTypes.NUMERIC_PRIMITIVE_TYPES) {
```

Comment:
To be compatible with postgres, it seems we just need to signatures, one with `double->double` and one with `numeric->numeric`.
From: https://www.postgresql.org/docs/current/functions-math.html
```
sign ( numeric ) → numeric
sign ( double precision ) → double precision
```
Tested locally:
```
matriv=> select pg_typeof(sign(1::bigint));
    pg_typeof     
------------------
 double precision
(1 row)

matriv=> select pg_typeof(sign(1::int));
    pg_typeof     
------------------
 double precision
(1 row)

matriv=> select pg_typeof(sign(1.2::float));
    pg_typeof     
------------------
 double precision
(1 row)

matriv=> select pg_typeof(sign(1.2::numeric));
 pg_typeof 
-----------
 numeric
(1 row)
```

Reply:
divided primitive types in two groups
BYTE, SHORT, INTEGER, LONG will return numeric value
DOUBLE, FLOAT will return double precision value

Reply:
![sign_test](https://github.com/crate/crate/assets/49018424/97e97b9a-e6a5-488d-8c2e-83c79ef13e8d)

The last commit should be correct